compiler/hash-typecheck/src/exhaustiveness/mod.rs

```diff
@@ -132,3 +135,32 @@ pub(crate) trait AccessToUsefulnessOps: AccessToStorage {
 }
 
 impl<T: AccessToStorage> AccessToUsefulnessOps for T {}
+
+/// Wraps a type `T` in a structure that contains information to be able to
+/// format `T` using [TcFormatter].
+///
+/// This can wrap any type, but only types that have corresponding `fmt_*`
+/// methods in [TcFormatter] are useful with it.
+pub struct PatForFormatting<'tc, T> {
```

Comment:
Why is this needed?

Reply:
Since `exhaustiveness_store` and `global_storage` are now split, but the formatting for types doesn't interact with `exhaustiveness_storage`

Reply:
Why is exhaustiveness_store separate from global_storage?

Reply:
I didn't think it made sense putting `DeconstructedCtor` and `DeconstructedPat` into `hash_types` since these types are implementation detail for the typechecker exhaustiveness checker.